### ISSUE TITLE
Trendline not showing when using Chart.js Object Data Structure v2.1.2

### ISSUE THREAD

**Issue by gr3enk** [Makanz/chartjs-plugin-trendline#93]
The trendline isn't working when using [Chart.js Object Data Structure](https://www.chartjs.org/docs/latest/general/data-structures.html). 

Chart.js: v4.4.2
trendline-plugin: v2.1.2 ( Same problem described in Issue #74. But the Problem still occurs in the current version )

The following Dataset configuration is not working:
```
const datasets: [{
	data: [
            { x: '202309', y: 144214.8 },
            { x: '202310', y: 144514.8 },
            { x: '202311', y: 150626 },
            { x: '202312', y: 204523 },
            { x: '202401', y: 125402 },
            { x: '202402', y: 225232 },
            { x: '202403', y: 2786435 },
            { x: '202404', y: 584833 },
        ],
	label: 'Usage',
	yAxisID: 'y',
	lineTension: 0,
	borderWidth: 2,
	trendlineLinear: {
		colorMin: '#ff0000',
		colorMax: '#00ff00',
		lineStyle: 'dotted',
		width: 2,
		yAxisKey: 'y',
		xAxisKey: 'x',
		projection: true,
	},
}, ],
```

This one works:
```
const datasets: [{
	data: [144214.8, 144514.8, 150626, 204523, 125402, 225232, 2786435, 584833],
	label: 'Usage',
	yAxisID: 'y',
	lineTension: 0,
	borderWidth: 2,
	trendlineLinear: {
		colorMin: '#ff0000',
		colorMax: '#00ff00',
		lineStyle: 'dotted',
		width: 2,
		yAxisKey: 'y',
		xAxisKey: 'x',
		projection: true,
	},
}, ],
```

Here is the chart config used for both examples:
```
const chartConfig = {
	type: 'line',
	data: {
		datasets: datasets,
		labels: ['202309', '202310', '202311', '202401', '202402', '202403', '202404'],
	},
	options: {
		scales: {
			x: {
				ticks: {
					padding: 8
				},
			},
			y: {
				type: 'linear',
				display: true,
				position: 'left',
				title: {
					display: true,
					text: 'Usage',
					padding: 8,
				},
			},
		},
	},
	plugins: [pluginTrendlineLinear],
};
```

**Comment by Makanz** [comment]
I've made some changes, and in my tests, both of your examples worked. Please try the latest version I've just released and let me know if you encounter any issues. Version 2.1.3

**Comment by gr3enk** [comment]
Fixed with Version 2.1.3
